lib/versioning/VersioningRequestProcessor.js

```diff
@@ -255,19 +255,13 @@ class VersioningRequestProcessor {
         // - versioning && !versionId: create a new version
         // - !versioning && versionId: update (PUT/DELETE) an existing version
         //                             if versionId === '' update master version
-        // other cases are invalid and metadata does nothing
+        // - versioning && versionId: create a new version which has
+        //   the specified versionId (e.g. for async replication)
```

Comment:
But as we were discussing, I think the extension of the behavior is OK and doesn't impact existing operations in S3.